test/cql/varchar_as_table_name_test.cql

```diff
@@ -0,0 +1,4 @@
+CREATE KEYSPACE reproducer WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1};
```

Comment:
Please add a comment explaining that this is a regression test for #10642. Simple e.g.
```
-- regression test for #10642
```